Engine/Visa.cs

```diff
@@ -4,746 +4,1043 @@
+        private delegate int viUnlockDelegate(int vi);
+
+        private static viOpenDefaultRMDelegate viOpenDefaultRMRef;
+        private static viFindRsrcDelegate viFindRsrcRef;
+        private static viFindNextDelegate viFindNextRef;
+        private static viParseRsrcDelegate viParseRsrcRef;
+        private static viParseRsrcExDelegate viParseRsrcExRef;
+        private static viOpenDelegate viOpenRef;
+        private static viCloseDelegate viCloseRef;
+        private static viGetAttributeDelegate1 viGetAttribute1Ref;
+        private static viGetAttributeDelegate2 viGetAttribute2Ref;
+        private static viGetAttributeDelegate3 viGetAttribute3Ref;
+        private static viSetAttributeDelegate1 viSetAttribute1Ref;
+        private static viSetAttributeDelegate2 viSetAttribute2Ref;
+        private static viStatusDescDelegate viStatusDescRef;
+        private static viEnableEventDelegate viEnableEventRef;
+        private static viDisableEventDelegate viDisableEventRef;
+        private static viInstallHandlerDelegate viInstallHandlerRef;
+        private static viUninstallHandlerDelegate viUninstallHandlerRef;
+        private static viReadDelegate viReadRef;
+        private static viWriteDelegate viWriteRef;
+        private static viReadSTBDelegate viReadSTBRef;
+        private static viClearDelegate viClearRef;
+        private static viLockDelegate viLockRef;
+        private static viUnlockDelegate viUnlockRef;
+        
+        static TraceSource staticLog = OpenTap.Log.CreateSource("Visa");
+        static Visa()
+        {
+            object instance = null;
+            var visaProviders = TypeData.GetDerivedTypes<IVisaProvider>();
+            var instances = new SortedDictionary<double, (string, IVisa)>();
+            foreach (var typeDataIVisa in visaProviders) 
+            {
+                if (typeDataIVisa != null)
+                {
+                    try 
+                    {
+                        staticLog.Debug("Attempting to load IVisaProvider: {0}", typeDataIVisa.Name);
+                        IVisaProvider providerInstance = (IVisaProvider)typeDataIVisa.CreateInstance();
+                        if (providerInstance.Visa != null)
+                        {
+                            instances[providerInstance.Order] = (typeDataIVisa.Name, providerInstance.Visa); 
+                            staticLog.Debug("Loaded IVisaProvider: {0}", typeDataIVisa.Name);
+                        }
+                    }
+                    catch {}
+                }            
+            }
+            if (instances.Count == 0)
+            {
+                throw new Exception("IVisaProvider unable to load VISA");
+            }
+            
+            instance = instances.Values.First().Item2;
+            staticLog.Debug("Using IVisaProvider: {0}", instances.Values.First().Item1);
```

Comment:
maybe we dont need this if we order a bit earlier. In the default use case it should just say "Loaded IVisaprovider  X" and nothing else.

Reply:
See responses on comments above. I don't see a path to reorder differently.